pkg/remediator/watch/filteredwatcher.go

```diff
@@ -73,6 +73,8 @@ type Runnable interface {
 	Stop()
 	Run(ctx context.Context) status.Error
 	ClearManagementConflictsWithKind(gk schema.GroupKind)
+	SetLatestCommitForGVK(string)
+	latestCommitForGVK() string
```

Comment:
If the setter's public, might as well make the getter public too.  Private methods on interfaces are rarely a good idea, especially public interfaces.

Reply:
The interface is only used within the package, so the whole interface could be private TBH. Do you have a style reference on why private methods on interfaces are a bad idea?

Updated to only use public methods.

Reply:
If the interface is only used in the package, then the interface can be private. If the interface is private, then there's no extra value from making the interface methods private. Making them private is meaningless. They might as well just all be public.

If the interface is public, then all the interface methods should be public, or else it can't be implemented or used by anything outside of the package.

So AFAICT, there's no reason to make private interface methods at all.